numba/targets/builtins.py

```diff
@@ -49,6 +47,36 @@ def generic_is(context, builder, sig, args):
     else:
         return cgutils.false_bit
 
+
+@lower_builtin('==', types.Any, types.Any)
+def generic_eq(context, builder, sig, args):
+    """
+    Default implementation for `x == y` that fallback to `is`
```

Comment:
Added check for infinite recursion in https://github.com/numba/numba/pull/1865/commits/d40944d00169741993b1b1cf6da5fae1c1058813